### ISSUE TITLE
Upstream service config has no expose config available unlike what is documented

### ISSUE THREAD

**Issue by nitsh** [hashicorp/consul#7353]
#### Overview of the Issue

The docs for upstream config of sidecar proxy suggests that there is a feature to support 'expose' config but there doesn't seem to be any such feature implemented as it fails the validation : https://www.consul.io/docs/connect/registration/service-registration.html#expose-1

#### Reproduction Steps

Steps to reproduce this issue :

1. Create a file `service.json` with following config 
```
{
  "services": [
    {
      "id": "client-service",
      "name": "client-service",
      "port": 443,
      "connect": {
        "sidecar_service": {
          "proxy": {
            "upstreams": [
              {
                "destination_name": "upstream-service",
                "local_bind_port": 13010,
                "expose": {
                  "paths": [
                    {
                      "path": "/healthcheck",
                      "protocol": "http",
                      "local_path_port": 8080,
                      "listener_port": 443
                    }
                  ]
                }
              }
            ]
          }
        }
      }
    }
  ]
}
```
2. Validate the config with 
```
docker run -it -v `pwd`:`pwd`  consul:1.6.2 consul validate `pwd`/service.json
```
```
Config validation failed: Error parsing .../service.json: 1 error occurred:
	* invalid config key services[0].connect.sidecar_service.proxy.upstreams[0].expose
```

**Comment by blake** [comment]
Hi @nitsh,

The particular `expose` reference you linked is a bit confusing as it gives the impression that this option should be placed under `upstreams`. Looking at the code it seems the expose configuration can only be specified under `service[s].connect.proxy` ([agent/config/config.go#L92-L93](https://github.com/hashicorp/consul/blob/v1.6.2/agent/config/config.go#L92-L93)) or `service[s].connect.sidecar_service.proxy` (see [agent/config/config.go#L104-L105](https://github.com/hashicorp/consul/blob/v1.6.2/agent/config/config.go#L104-L105)). The [Upstream](https://github.com/hashicorp/consul/blob/v1.6.2/agent/config/config.go#L483-L518) struct does not list `Expose` as a valid config field.

The following, however, should be a valid service config.

```json
{
  "services": [
    {
      "id": "client-service",
      "name": "client-service",
      "port": 443,
      "connect": {
        "sidecar_service": {
          "proxy": {
            "upstreams": [
              {
                "destination_name": "upstream-service",
                "local_bind_port": 13010
              }
            ],
            "expose": {
              "paths": [
                {
                  "path": "/healthcheck",
                  "protocol": "http",
                  "local_path_port": 8080,
                  "listener_port": 443
                }
              ]
            }
          }
        }
      }
    }
  ]
}
```

**Comment by nitsh** [comment]
@blake Yes. I wanted to raise an issue regarding that confusion in that documentation link. If it's not expected behavior, maybe the documentation should be updated accordingly.

**Comment by blake** [comment]
@nitsh Thank you for reporting this issue. The documentation was corrected in #7515, so I am closing this issue.
